pkg/provider/vault/auth.go

```diff
@@ -41,9 +41,9 @@ const (
 
 // setAuth gets a new token using the configured mechanism.
 // If there's already a valid token, does nothing.
-func (c *client) setAuth(ctx context.Context, cfg *vault.Config) error {
+func (c *client) setAuth(ctx context.Context, spec *esv1beta1.VaultProvider, cfg *vault.Config) error {
 	// Switch to auth namespace if different from the provider namespace
-	restoreNamespace := c.useAuthNamespace(ctx)
+	restoreNamespace := c.useAuthNamespace(ctx, spec)
```

Comment:
```suggestion
	if c.store.Namespace != nil {  //set namespace before checking the need for AuthNamespace
		ns = *c.store.Namespace
       }
       restoreNamespace := c.useAuthNamespace(ctx)
```

Reply:
I think this one will solve it @Skarlso

Reply:
Nice. I'm back from vacation today and will take a look at it in the evening.

Reply:
@gusfcarvalho Ah, but this won't work because it needs to restore the previous namespace which you deleted here completely. 🤔

Reply:
@gusfcarvalho That's not the right namespace. We are looking for the spec namespace here not the store namespace.

Reply:
Store is the vault spec 😅 there is no difference between them, right? I mean they are literally the same struct strictly speaking, no?

Reply:
I mean “c.store” here is of type VaultProvider.

Reply:
So c.store.namespace is the same as vaultSpec.namespace, and the code is still the same, nov

Reply:
What. Store is the whole store and spec is the vault provider spec. Store namespace is the store namespace spec namespace is the vault namespace.

Reply:
It's not  - this is the [vault client definition](https://github.com/external-secrets/external-secrets/blob/main/pkg/provider/vault/client.go#L40-L51)
```
type client struct {
	kube      kclient.Client
	store     *esv1beta1.VaultProvider // Its not the store, its the Vault Spec
	log       logr.Logger
	corev1    typedcorev1.CoreV1Interface
	client    util.Client
	auth      util.Auth
	logical   util.Logical
	token     util.Token
	namespace string
	storeKind string
}

```

Reply:
so adding vaultSpec instead of the client store isnt going to change a thing IMO - so I'd expect the bug to still exist. Which is why my proposed fix was just instantiating the namespace before calling `useAuthNamespace` - as there is currently a condition where if `namespace` is set but `auth.namespace` isnt, we are not using `namespace` at all.

Reply:
Omg okay so store is the provider Jesus. 🤣 sorry I missed that. 

But the problem with the set back thing was that it was setting back to an empty string if that value was set it ignored it. I added a unit test that reproduced that behaviour. 🤔

But I'll try what you're doing. So you think the reset logic is fine as is?

Reply:
Oh now I see what you mean ah. So it was literally checking the same thing with the spec as well. The odd thing is the unit test failing then. I'll have to look into it a bit more.

Reply:
@gusfcarvalho Are you happy with the changes? :)